### PR TITLE
feat(core): update mfa in sign in experience

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add "mfa" to the api `/sign-in-exp` in both `GET` and `PATCH`.

## Testing

Auto tests in the repo.

## Checklist

- [x] `.changeset` will do it after this whole feature is completed
- [x] unit tests
- [x] integration tests
- [ ] docs

OR

- [ ] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/core/src/libraries/sign-in-experience/mfa.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { MfaFactor, type Mfa } from '@logto/schemas';
+
+import assertThat from '#src/utils/assert-that.js';
+
+export const validateMfa = (mfa: Mfa) => {
+  const backupCodeEnabled = mfa.factors.includes(MfaFactor.BackupCode);
+
+  if (backupCodeEnabled) {
+    assertThat(mfa.factors.length > 1, 'sign_in_experiences.backup_code_cannot_be_enabled_alone');
+  }
```
Comment (simeng-li) [inline packages/core/src/libraries/sign-in-experience/mfa.ts:10]: nit: Double check, is this the only validation logic we need?

File: packages/schemas/src/foundations/jsonb-types.ts [hunk 2]
```diff
@@ -164,6 +164,28 @@ export const customContentGuard = z.record(z.string());
 
 export type CustomContent = z.infer<typeof customContentGuard>;
 
+export enum MfaFactor {
+  TOTP = 'TOTP',
+  WebAuthn = 'WebAuthn',
+  BackupCode = 'BackupCode',
+}
+
+export const mfaFactorsGuard = z.nativeEnum(MfaFactor).array();
```
Comment (simeng-li) [inline packages/schemas/src/foundations/jsonb-types.ts:173]: Can we set multiple MAPFactors of the same type? e.g.  Two backup codes or three TOTP?

Comment (wangsijie) [reply packages/schemas/src/foundations/jsonb-types.ts]: The user can have multiple factors, but here is for feature control, so multiple with same type makes no sense.

Updated with a new check.

**Review by xiaoyijun — APPROVED** [review]
LGTM

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**